pipeline/id_minter/src/test/scala/weco/pipeline/id_minter/database/IdentifiersDaoTest.scala

```diff
@@ -82,6 +82,73 @@ class IdentifiersDaoTest
       }
     }
 
+    it("raises an exception if an unexpected id is returned")  {
+      val testCases = Table(
+        "identifierType",
+        IdentifierType.MiroImageNumber,
+        IdentifierType.SierraSystemNumber,
+        IdentifierType.CalmRecordIdentifier
+      )
+      forAll(testCases){
+        identifierType =>
+          val sourceIdentifier = createSourceIdentifierWith(value="deadbeef", identifierType=identifierType)
+          val identifier = createSQLIdentifierWith(
+            sourceIdentifier = sourceIdentifier
+          )
+          val badCaseIdentifier = sourceIdentifier.copy(value="DEADBEEF")
+          withIdentifiersDao(existingEntries = Seq(identifier)) {
+            case (identifiersDao, _) =>
+              val triedLookup = identifiersDao.lookupIds(List(badCaseIdentifier))
+              triedLookup match {
+                case Failure(e) => e.getMessage.strip shouldBe s"The query returned a sourceIdentifier (Work[${identifierType.id}/deadbeef]) which we weren't looking for (List(Work[${identifierType.id}/DEADBEEF]))"
+                case unexpected => fail(s"lookupIds should have failed: $unexpected")
+              }
+              triedLookup shouldBe a[Failure[_]]
```

Comment:
Move this line above the pattern match, then you can do `triedLookup.failed.get`?